packages/back-end/src/controllers/experiments.ts

```diff
@@ -1707,12 +1705,50 @@ export async function postSnapshot(
 
   req.checkPermissions("runQueries", experiment.project || "");
 
-  let { statsEngine } = req.body;
+  let project = null;
+  if (experiment.project) {
+    project = await findProjectById(experiment.project, org.id);
+  }
+  const orgSettings: OrganizationSettings = org.settings as OrganizationSettings;
+  const { settings } = getScopedSettings({
+    organization: org,
+    project: project ?? undefined,
+    experiment,
+  });
+  const statsEngine = settings.statsEngine.value;
+  const { phase, dimension } = req.body;
+
+  const allExperimentMetricIds = uniq([
+    ...experiment.metrics,
+    ...(experiment.guardrails ?? []),
+  ]);
+  const allExperimentMetrics = await Promise.all(
+    allExperimentMetricIds.map((m) => getExperimentMetricById(m, org.id))
+  );
+  const denominatorMetricIds = uniq<string>(
+    allExperimentMetrics
+      .map((m) => m?.denominator)
+      .filter((d) => d && typeof d === "string") as string[]
+  );
+  const denominatorMetrics = (
+    await Promise.all(denominatorMetricIds.map((m) => getMetricById(m, org.id)))
+  ).filter(Boolean) as MetricInterface[];
+  const datasource = await getDataSourceById(experiment.datasource, org.id);
+
   const {
     metricRegressionAdjustmentStatuses,
     regressionAdjustmentEnabled,
-  } = req.body;
-  const { phase, dimension } = req.body;
+  } = getAllMetricRegressionAdjustmentStatuses({
+    allExperimentMetrics,
+    denominatorMetrics,
+    orgSettings,
+    statsEngine,
+    experimentRegressionAdjustmentEnabled:
+      experiment.regressionAdjustmentEnabled,
+    experimentMetricOverrides: experiment.metricOverrides,
+    datasourceType: datasource?.type,
+    hasRegressionAdjustmentFeature: true,
```

Comment:
Is this hard coded because the front-end is already responsible for the check?

Reply:
mostly yeah. Subjective UX decision, but I figure if they've somehow gotten to the state where CUPID could have been enabled and then they lose support somehow (license expired), we don't immediately want to change their backend analysis based on pre-existing frontend changes.